### ISSUE TITLE
PR companion is not deploying images used in live samples

### ISSUE THREAD

**Issue by OnkarRuikar** [mdn/fred#964]
### Summary

Related to `mdn/content` PR  https://github.com/mdn/content/pull/28805

PR companion for `mdn/content` doesn't deploy images to the `BUILD_LIVE_SAMPLES_BASE_URL` server.

### URL

https://github.com/mdn/content/pull/28805#issuecomment-1696881023

### Reproduction steps

1. Commit an image (say `demo.jpg`) in a local branch and use it in a live sample for the page using HTML `<img src=demo.jpg>`
2. Submit a PR in `mdn/content` repo.
3. Open the preview link for the page generated by the PR companion.
4. In the live sample the image is not rendered.

In this case the image was not deployed to `https://live.mdnyalp.dev/en-US/docs/Web/CSS/aspect-ratio/plumeria.jpg`. Once the changes go live it will be deployed to `https://live.mdnyalp.dev` so this won't be reproducible using the PR.

### Expected behavior

On PR preview pages live samples with images should render the images using relative paths `demo.jpg` or `./demo.jpg`.

### Actual behavior

Images committed along with markdown pages aren't rendering in live sample outputs.

### Device

Desktop

### Browser

Firefox

### Browser version

Stable

### Operating system

Linux

### Screenshot

![_1](https://github.com/mdn/fred/assets/87750369/d65d360f-fce0-4fbc-b750-46cc55f2f2f2)

### Validations

- [X] I have read the [Community Participation Guidelines](https://www.mozilla.org/en-US/about/governance/policies/participation/).
- [X] I have verified that there isn't [already an issue](https://github.com/mdn/yari/issues) that reports the same bug to avoid creating a duplicate.
- [X] I have checked that this is a concrete bug. For Q&A open a [GitHub Discussion](https://github.com/orgs/mdn/discussions/categories/platform).

**Comment by OnkarRuikar** [comment]
Note, the preview links in the content PR will work for a limited time. So plan the investigation accordingly.

**Comment by yin1999** [comment]
The assets (only on `main` branch) would only be deployed when a [stage build](https://github.com/mdn/yari/blob/231d6aab8f1c8efe159d268c261446c5b7ae12d9/.github/workflows/stage-build.yml) is completed. So this would prevents us from fetching the images. Actually, the image does deployed with the main page's domain: `https://pr28805.content.dev.mdn.mozit.cloud/en-US/docs/Web/CSS/aspect-ratio/plumeria.jpg`.

The current live sample uses [`Message` event handler to pass the code of the main page to the live sample iframe](https://github.com/mdn/yari/blob/231d6aab8f1c8efe159d268c261446c5b7ae12d9/client/src/document/hooks.ts#L24-L70) instead of directly generating a static page with related assets. I don’t know if this is the reason why the developers only use the single domain to deploy the live samples for PR previews.

**Comment by OnkarRuikar** [comment]
> Actually, the image does deployed with the main page's domain: https://pr28805.content.dev.mdn.mozit.cloud/en-US/docs/Web/CSS/aspect-ratio/plumeria.jpg.

I think changing [the environment variable](https://github.com/mdn/content/blob/81fc9ae4093636637e4ea7356edf9c3807594d42/.github/workflows/pr-test.yml#L90C4-L90C4) from
```yml
BUILD_LIVE_SAMPLES_BASE_URL: https://live.mdnyalp.dev
```
to
```yml
BUILD_LIVE_SAMPLES_BASE_URL: https://pr<prNo>.content.dev.mdn.mozit.cloud
```
and changing [the macro](https://github.com/mdn/yari/blob/231d6aab8f1c8efe159d268c261446c5b7ae12d9/kumascript/macros/LiveSampleURL.ejs#L15C67-L15C67) like following:
```diff
-- let samplePath = `${pagePath}/runner.html`.replace(/\/\/+/g, "/");
++ let samplePath = `${baseUrl}/runner.html`.replace(/\/\/+/g, "/");
```
may work.

**Comment by estelle** [comment]
This is currently (May 8, 2024) an issue in the following PRs:

* https://github.com/mdn/content/pull/33370 (closed, but see [comment](https://github.com/mdn/content/pull/33370#discussion_r1587088211)
* https://github.com/mdn/content/pull/33461
* https://github.com/mdn/content/pull/31713

**Comment by caugner** [comment]
@OnkarRuikar @yin1999 @estelle Is this still an issue?

**Comment by OnkarRuikar** [comment]
> [@OnkarRuikar](https://github.com/OnkarRuikar) [@yin1999](https://github.com/yin1999) [@estelle](https://github.com/estelle) Is this still an issue?

@caugner yes: https://pr39990.review.mdn.allizom.net/en-US/docs/Web/CSS/aspect-ratio#fallback_to_natural_aspect_ratio

**Comment by caugner** [comment]
Got it:

- The PR adds a _new_ image (here: `trex.png`).
- The image is used in a live sample.
- The live sample is rendered in the Playground (using the production deployment: `mdnplay.dev`).
- The Playground doesn't serve the image.

Note: I checked, and the image is properly deployed to the review bucket.

As the `runner.html` request gets the review host as a referrer, the Playground could potentially use that to fetch the image from the right path. However, our production deployment doesn't fetch from the review bucket, and it shouldn't fetch the image from the review host either.

So maybe the best solution here is to add separate runner hosts for the review environment, to allow this environment-specific behavior.

**Comment by OnkarRuikar** [comment]
Why does https://pr39990.review.mdn.allizom.net/en-US/docs/Web/CSS/aspect-ratio/trex.png redirect to
https://developer.mozilla.org/en-us/docs/web/css/aspect-ratio/trex.png ?

**Comment by caugner** [comment]
> Why does [pr39990.review.mdn.allizom.net/en-US/docs/Web/CSS/aspect-ratio/trex.png](https://pr39990.review.mdn.allizom.net/en-US/docs/Web/CSS/aspect-ratio/trex.png) redirect to
> [developer.mozilla.org/en-us/docs/web/css/aspect-ratio/trex.png](https://developer.mozilla.org/en-us/docs/web/css/aspect-ratio/trex.png) ?

This is a feature that is [specific to the Review environment](https://github.com/mdn/yari/blob/03d394862272e8714cd3219432e67e66da29dd82/cloud-function/src/handlers/proxy-content-assets.ts#L43-L52), and allows pages (afaik especially translated ones) to reference _existing_ images that weren't deployed together with the modified pages.

The lowercasing is not intended, but it doesn't make a difference in this case.
